draft-ietf-suit-update-management.md

```diff
@@ -177,6 +179,121 @@ suit-wait-event-day-of-week-utc | uint | Wait until days since Sunday UTC
 
 suit-wait-event-other-device-version reuses the encoding of suit-parameter-version-match. It is encoded as a sequence that contains an implementation-defined bstr identifier for the other device, and a list of one or more SUIT_Parameter_Version_Match.
 
+## suit-parameter-component-metadata {#suit-parameter-component-metadata}
+
+In some instances, a system may need to know the file metadata for a component. This metadata can include:
+
+* creator
+* creation time
+* modification time
+* default permissions (rwx)
+* a map of user/permission pairs
+* a map of role/permission pairs
+* a map of group/permission pairs
+* file type
+
+Component metadata is applied at time of fetch, copy, or write; see {{I-D.ietf-suit-manifest}}, sections 8.4.10.4, 8.4.10.5, 8.4.10.6. Therefore, the component metadata parameter must be set in advance of the component being fetched, copied into, or written.
+
+ 
+### Creator {#suit-meta-creator}
+
+Sometimes, management of file systems requires that the creator of each file is correctly recorded. Because the default creator of files will be the update agent, this can obscure the actual creator of each file. The Creator metadata element allows overriding the default behaviour and setting the correct creator.
+
+The creator is defined as follows:
+
+~~~ CDDL
+SUIT_meta_actor_id = UUID_Tagged / bstr / str / int
+UUID_Tagged = #6.37(bstr)
+~~~
+
+The actor ID can be whatever is most appropriate for any given system. Typically, the actor ID will be a string (e.g., username), integer (e.g., POSIX userid), or UUID (e.g., TEEP TA UUID).
+
+
+### Creation & Modification Time
+
+The creation and modification times are defined by CBOR time types. These are defined in {{RFC8949}}, Section 3.4.2. The CBOR tag is REQUIRED when either creation or modification time are provided.
+
+~~~ CDDL
+suit-meta-modification-time => #6.1(uint)
+suit-meta-creation-time => #6.1(uint)
+~~~
+
+### Component Default Permissions
+
+Typical permissions management systems require read, write, and execute permissions that are applied to all users who do not have their own explicit permissions. These are the default permissions for the current component. Default permissions are described by the following CDDL:
+
+~~~ CDDL
+SUIT_meta_permissions = uint .bits SUIT_meta_permission_bits
+SUIT_meta_permission_bits = &(
+    r: 2, w: 1, x: 0,
+    * $$SUIT_meta_permission_bits_extensions
```

Comment:
This set seems very POSIX centric.  Compare https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/cc732880(v=ws.10)
For example, the "modify" bit is separate from the "write" bit and allows deletion of the file.
Similarly, the "change permissions" bit allows changing others' permissions.
Of course the CDDL does allow extensions, so it may be ok.